### ISSUE TITLE
[tune] Pattern for picking best results from successive tune runs?

### ISSUE THREAD

**Issue by ca-scribner** [ray-project/ray#7350]
I'm building an image classification pipeline that will maintain itself over time.  Schematically it is something like:

- images stream in and are normalized/stored over time
- every so often, a new tune hyperparameter training run is triggered, resulting in new model params
- a governor process inspects the available hyperparameter results (possibly from most recent tune run, or maybe from a previous tune run), finds the best configuration, and if that is not currently deployed it trains/deploys that config

Are there any best practice patterns for doing this with tune?

My current thoughts were:

- store persistent analysis results from each tune.run centrally (eg: set local_dir or remote_dir to some global location), then have the governor search that location for best results by instantiating an ExperimentAnalysis for each experiment_state.json  
- store persistent analysis results from each tune.run somewhere (not necessarily in a single ray_results dir structure), and keep a DB knowing where.  Governor can then query the DB for path to experiment_state.json
  - At this point, I could also just store results and param json to the DB and ditch the tune analysis results entirely

Are there better patterns available?  This feels fine for a single tune analysis, but I'll have many for competing models, etc., that I'll want to search across.  I'll also want to be able to deprecate these selectively if I update the training data, etc.

**Comment by richardliaw** [comment]
I would probably just ditch the analysis object for now and build something specialized for my DB. 

I would also implement a specific logger class for that said DB, so that I don't need to wrangle csv or json files.

**Comment by ca-scribner** [comment]
Thanks for the suggestions, that makes sense.

For a logger class, you mean to implement something inheriting the Tune Logger class and specified to Tune by tune.run(loggers=MyLoggers)?  

Thanks to your suggestion I also learned some new things from the docs.  Hadn't noticed before there's an MLFlow logger available.  Might try using both that and custom and just see which feels best.
